opensrp-giz-malawi/src/main/java/org/smartregister/giz/dao/ReportDao.java

```diff
@@ -0,0 +1,419 @@
+package org.smartregister.giz.dao;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.text.TextUtils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.Days;
+import org.smartregister.CoreLibrary;
+import org.smartregister.dao.AbstractDao;
+import org.smartregister.domain.Alert;
+import org.smartregister.domain.AlertStatus;
+import org.smartregister.domain.jsonmapping.Location;
+import org.smartregister.domain.jsonmapping.util.LocationTree;
+import org.smartregister.domain.jsonmapping.util.TreeNode;
+import org.smartregister.giz.application.GizMalawiApplication;
+import org.smartregister.giz.domain.EligibleChild;
+import org.smartregister.giz.domain.VillageDose;
+import org.smartregister.immunization.db.VaccineRepo;
+import org.smartregister.immunization.domain.Vaccine;
+import org.smartregister.immunization.domain.VaccineCondition;
+import org.smartregister.immunization.domain.VaccineSchedule;
+import org.smartregister.immunization.domain.VaccineTrigger;
+import org.smartregister.immunization.domain.jsonmapping.Condition;
+import org.smartregister.immunization.domain.jsonmapping.Due;
+import org.smartregister.immunization.domain.jsonmapping.Expiry;
+import org.smartregister.immunization.domain.jsonmapping.Schedule;
+import org.smartregister.immunization.domain.jsonmapping.VaccineGroup;
+import org.smartregister.immunization.repository.VaccineRepository;
+import org.smartregister.immunization.util.VaccinatorUtils;
+import org.smartregister.repository.EventClientRepository;
+import org.smartregister.util.AssetHandler;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import timber.log.Timber;
+
+public class ReportDao extends AbstractDao {
+
+    private static HashMap<String, HashMap<String, VaccineSchedule>> vaccineSchedules;
+    @SuppressLint("StaticFieldLeak")
+    private static Context context;
```

Comment:
You can use a weakReference here and add a getter for the Context to avoid suppressing `StaticFieldLeak`.
```suggestion
    private static WeakReference<Context> contextWeakReference;
```